tests/integration/common-operator-integ-suite.sh

```diff
@@ -14,87 +14,141 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# To be able to run this script it's needed to pass the flag --ocp or --kind
-set -eu -o pipefail
+# To be able to run this script, it's needed to pass the flag --ocp or --kind
+set -eux -o pipefail
+
+check_arguments() {
+  if [ $# -eq 0 ]; then
+    echo "No arguments provided"
+    exit 1
+  fi
+}
 
-if [ $# -eq 0 ]; then
-  echo "No arguments provided"
-  exit 1
-fi
+parse_flags() {
+  while [ $# -gt 0 ]; do
+    case "$1" in
+      --ocp)
+        shift
+        OCP=true
+        ;;
+      --kind)
+        shift
+        OCP=false
+        ;;
+      *)
+        echo "Invalid flag: $1"
+        exit 1
+        ;;
+    esac
+  done
+
+  if [ "${OCP}" == "true" ]; then
+    echo "Running on OCP"
+  else
+    echo "Running on kind"
+  fi
+}
 
-while [ $# -gt 0 ]; do
-  case "$1" in
-    --ocp)
-      shift
-      OCP=true
-      ;;
-    --kind)
-      shift
-      OCP=false
-      ;;
-    *)
-      echo "Invalid flag: $1"
-      exit 1
-      ;;
-  esac
-done
-
-if [ "${OCP}" == "true" ]; then
-  echo "Running on OCP"
-else
-  echo "Running on kind"
-fi
+initialize_variables() {
+  WD=$(dirname "$0")
+  WD=$(cd "$WD" || exit; pwd)
 
-WD=$(dirname "$0")
-WD=$(cd "$WD" || exit; pwd)
+  NAMESPACE="${NAMESPACE:-istio-operator}"
+  DEPLOYMENT_NAME="${DEPLOYMENT_NAME:-istio-operator}"
+  CONTROL_PLANE_NS="${CONTROL_PLANE_NS:-istio-system}"
+  COMMAND="kubectl"
 
-NAMESPACE="${NAMESPACE:-istio-operator}"
-DEPLOYMENT_NAME="${DEPLOYMENT_NAME:-istio-operator}"
-CONTROL_PLANE_NS="${CONTROL_PLANE_NS:-istio-system}"
-COMMAND="kubectl"
+  if [ "${OCP}" == "true" ]; then
+    COMMAND="oc"
+  fi
 
-if [ "${OCP}" == "true" ]; then
-  COMMAND="oc"
-fi
+  echo "Using command: ${COMMAND}"
 
-BRANCH="${BRANCH:-maistra-3.0}"
+  BRANCH="${BRANCH:-maistra-3.0}"
 
-if [ "${OCP}" == "true" ]; then
-  ISTIO_MANIFEST="${WD}/../../config/samples/istio-sample-openshift.yaml"
-else
-  ISTIO_MANIFEST="${WD}/../../config/samples/istio-sample-kubernetes.yaml"
-fi
+  if [ "${OCP}" == "true" ]; then
+    ISTIO_MANIFEST="${WD}/../../config/samples/istio-sample-openshift.yaml"
+  else
+    ISTIO_MANIFEST="${WD}/../../config/samples/istio-sample-kubernetes.yaml"
+  fi
 
-TIMEOUT="3m"
+  TIMEOUT="3m"
 
-check_ready() {
-    local NS=$1
-    local POD_NAME=$2
-    local DEPLOYMENT_NAME=$3
+  # Set a default value for a variable
+  : "${RUN_ONLY_TEST:="false"}"
```

Comment:
If you change the assignment, perhaps it's more clear to the reader, making the comment above not necessary.
e.g. `RUN_ONLY_TEST=${RUN_ONLY_TEST:-"false"}

Although in my opinion this variable should be called `SKIP_BUILD` and should be passed via command line arguments (e.g. `--skip-build`).

Reply:
Added